syft_tensorflow/hook/hook.py

```diff
@@ -282,9 +286,35 @@ def _add_methods_from_native_tensor(tensor_type: type, syft_type: type):
                 # Add this method to the TF tensor
                 setattr(tensor_type, attr, getattr(syft_type, attr))
 
+    def _add_methods_to_eager_tensor(self):
+      """
+      Add required TensorFlowTensor methods to EagerTensor.
+
+      When a user creates a tensor, e.g. with `tf.constant([1,2,3])`, the
+      type of the returned object is an `EagerTensor`.  EagerTensor is defined
+      in tensorflow and is a super class of tf.Tensor.  However, EagerTensor is
+      not importable so we cannot add the properties to it that we want.
+      So we do that here, which requires us to instantiate an instance of an
+      EagerTensor to then get reference to the type.
+
+      We do it this way for 2 reasons:
+
+      1. Avoid monkeypatching functions per instance (e.g. having to overwrite the
+         function every time we make a tensor)
+      2. Most dunder methods are actually looked up on the class itself, rather
+         than the instance, so monkeypatching the instance doesn't even work for
+         things like __repr__.
+      """
+
+      dummy = tf.constant(0)
```

Comment:
would still love to prevent doing this, are we sure it's not possible? if we're not sure I'd be cool with adding a TODO for that